dataland-frontend/tests/e2e/fixtures/eutaxonomy/non-financials/EuTaxonomyDataForNonFinancialsFixtures.ts

```diff
@@ -7,192 +7,114 @@ import {
-    ),
-    alignedActivities: valueOrUndefined(
-      generateArray(generateAlignedActivity, 1, 3, undefinedProbabilityOfFields),
-      undefinedProbabilityOfFields,
-    ),
-    enablingShareInPercent: valueOrUndefined(randomPercentageValue(), undefinedProbabilityOfFields),
-    transitionalShareInPercent: valueOrUndefined(randomPercentageValue(), undefinedProbabilityOfFields),
-  };
-}
+  /**
+   * Generates a random aligned activity
+   * @returns a random aligned activity
+   */
+  generateAlignedActivity(): EuTaxonomyAlignedActivity {
+    return {
+      ...this.generateActivity(),
+      substantialContributionToClimateChangeMitigationInPercent: this.randomPercentageValue(),
+      substantialContributionToClimateChangeAdaptionInPercent: this.randomPercentageValue(),
+      substantialContributionToSustainableUseAndProtectionOfWaterAndMarineResourcesInPercent:
+        this.randomPercentageValue(),
+      substantialContributionToTransitionToACircularEconomyInPercent: this.randomPercentageValue(),
+      substantialContributionToPollutionPreventionAndControlInPercent: this.randomPercentageValue(),
+      substantialContributionToProtectionAndRestorationOfBiodiversityAndEcosystemsInPercent:
+        this.randomPercentageValue(),
+      dnshToClimateChangeMitigation: this.randomYesNo(),
+      dnshToClimateChangeAdaption: this.randomYesNo(),
+      dnshToSustainableUseAndProtectionOfWaterAndMarineResources: this.randomYesNo(),
+      dnshToTransitionToACircularEconomy: this.randomYesNo(),
+      dnshToPollutionPreventionAndControl: this.randomYesNo(),
+      dnshToProtectionAndRestorationOfBiodiversityAndEcosystems: this.randomYesNo(),
+      minimumSafeguards: this.randomYesNo(),
+    };
+  }
 
-/**
- * Generates a single fixture for the eutaxonomy-non-financials framework
- * @param undefinedProbabilityOfFields the probability of an undefined value per field
- * @returns the generated fixture
- */
-export function generateEuTaxonomyDataForNonFinancials(
-  undefinedProbabilityOfFields?: number,
-): EuTaxonomyDataForNonFinancials {
-  const data: EuTaxonomyDataForNonFinancials = {};
-  data.general = generateEuTaxonomyWithBaseFields(undefinedProbabilityOfFields);
-  data.opex = generateEuTaxonomyPerCashflowType(
-    assertDefined(data.general.referencedReports),
-    undefinedProbabilityOfFields,
-  );
-  data.capex = generateEuTaxonomyPerCashflowType(
-    assertDefined(data.general.referencedReports),
-    undefinedProbabilityOfFields,
-  );
-  data.revenue = generateEuTaxonomyPerCashflowType(
-    assertDefined(data.general.referencedReports),
-    undefinedProbabilityOfFields,
-  );
-  return data;
+  /**
+   * Generates fake data for a single cash-flow type for the eutaxonomy-non-financials framework
+   * @returns the generated data
+   */
+  generateEuTaxonomyPerCashflowType(): EuTaxonomyDetailsPerCashFlowType {
+    return {
+      totalAmount: this.randomDataPoint(generateCurrencyValue(), faker.finance.currencyCode()),
```

Comment:
Why do we even have the custom method `generateIso4217CurrencyCode()` in our code, if the faker can already create a currecny code?

Reply:
I changed a name an kept it as a wrapper